### PR TITLE
planning: scenario manager: add a map to avoid traversing vector here…

### PR DESCRIPTION
**Pull request by jmtao**
… and there

### CODE REVIEW

File: modules/planning/scenarios/scenario_manager.cc [hunk 1]
```diff
@@ -294,6 +294,30 @@ bool ScenarioManager::SelectScenario(
...
+        first_encountered_overlap_map_[ReferenceLineInfo::STOP_SIGN] =
+            overlap.second;
+        break;
+      case ReferenceLineInfo::SIGNAL:
+        first_encountered_overlap_map_[ReferenceLineInfo::SIGNAL] =
+            overlap.second;
+        break;
+      case ReferenceLineInfo::PNC_JUNCTION:
+        first_encountered_overlap_map_[ReferenceLineInfo::PNC_JUNCTION] =
+            overlap.second;
```
Comment (xiaoxq) [inline modules/planning/scenarios/scenario_manager.cc]: Simply first_encountered_overlap_map_[overlap.first] = overlap.second; for these three types?

Comment (YajiaZhang) [reply modules/planning/scenarios/scenario_manager.cc]: I guess there might be some customized assignment coming.

Comment (xiaoxq) [reply modules/planning/scenarios/scenario_manager.cc]: Then might be like:

    switch (overlap.first) {
      case ReferenceLineInfo::STOP_SIGN:
      case ReferenceLineInfo::SIGNAL:
      case ReferenceLineInfo::PNC_JUNCTION:
          first_encountered_overlap_map_[overlap.first] = overlap.second;
          break;

Or put them to default block, and use cases to handle customized ones :)

Comment (YajiaZhang) [reply modules/planning/scenarios/scenario_manager.cc]: Sure, I'm fine with either way. Let @jmtao decide.

Comment (jmtao) [reply modules/planning/scenarios/scenario_manager.cc]: haha. good catch! fixed, amended

**Pull request merged**